### PR TITLE
fix(Creampies): Make it so identities are respected when creaming or being creamed

### LINKED ISSUE #18540: Cream pie popup thrower and thrown entities are mixed up
## Description

Instead of "X has been creamed by banana cream pie" it should be "X has been creamed by Joe Jenero" (person who threw pie)

image

**Reproduction**

**Screenshots**

Peek 2023-08-01 22-18

**Additional context**

### LINKED ISSUE #30676: Banana cream pies reveal identity
## Description
See title

**Reproduction**
- Give a mask to an Urist without an ID so they get an unknown identity
- Throw a banana cream pie at him
- The popup will say "Urist McHands ..."

**Screenshots**
Screenshot (219)

Screenshot (221)

**Additional context**
It should use the Name function in Content.Shared.IdentityManagement instead.
Good for a beginner PR.

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Made it so identities are respected when creaming or being creamed

## Why / Balance

Resolves #18540
Resolves #30676

## Media

# before (refer to)
#30676
#18540

# After
[Screencast from 08-05-2024 04:17:09 PM.webm](https://github.com/user-attachments/assets/b70689fb-e61d-4da6-90de-efbceef6ead7)

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by Brandon-Huu** [comment]
Code kinda shit IMO

**Comment by Brandon-Huu** [comment]
Might fix

**Comment by Brandon-Huu** [comment]
https://github.com/space-wizards/space-station-14/blob/f7d8b4e7d469c03f4adab33307af4a44e44b0319/Content.Server/Flash/FlashSystem.cs#L139-L143

No idea if I'm supposed to use Identity() or not.

File: Content.Server/Nutrition/EntitySystems/CreamPieSystem.cs [hunk 1]
```diff
@@ -90,13 +91,24 @@ private void ActivatePayload(EntityUid uid)
...
             var otherPlayers = Filter.Empty().AddPlayersByPvs(uid);
+
             if (TryComp<ActorComponent>(args.Target, out var actor))
-            {
                 otherPlayers.RemovePlayer(actor.PlayerSession);
+
+            if (args.Component.Thrower is null)
+            {
+            _popup.PopupEntity(Loc.GetString("cream-pied-component-on-hit-by-unkown-message"), uid, args.Target);
+            _popup.PopupEntity(Loc.GetString("cream-pied-component-on-hit-by-unknown-message-others", ("owner", args.Target)), uid, otherPlayers, false);
```
Comment (EmoGarbage404) [inline Content.Server/Nutrition/EntitySystems/CreamPieSystem.cs:102]: the "owner" param on this isn't passing in an identity.

File: Resources/Locale/en-US/nutrition/components/cream-pied-component.ftl [hunk 2]
```diff
@@ -1,2 +1,4 @@
 cream-pied-component-on-hit-by-message =  You have been creamed by {$thrower}!
-cream-pied-component-on-hit-by-message-others = {$owner} has been creamed by {$thrower}!
\ No newline at end of file
+cream-pied-component-on-hit-by-unkown-message =  You have been creamed!
```
Comment (EmoGarbage404) [inline Resources/Locale/en-US/nutrition/components/cream-pied-component.ftl:2]: "unkown"

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by Brandon-Huu** [comment]
Gyaat

**Comment by Brandon-Huu** [comment]
Mistakes where made

**Pull request closed without merging**